### ISSUE TITLE
Adhere the `msi` package to the target spec

### ISSUE THREAD

**Issue by spencergilbert** [vectordotdev/vector#12554]

**Comment by spencergilbert** [comment]
@jszwedko I suspect it's worth passing on this for now, given the benefits vs my getting up to speed on MSI/testing on windows/etc

**Comment by jszwedko** [comment]
Yeah, agreed, I think this is another we can deprioritize until we have more demand.
